src/Nethermind/Nethermind.Network.Discovery/RoutingTable/NodeTable.cs

```diff
@@ -128,6 +128,10 @@ public bool MoveNext()
                     return true;
                 }
             }
+            catch (NullReferenceException)
+            {
+                // TODO: investigate why this happens
```

Comment:
maybe log the internals of the enumerator here?

Reply:
Changed it to be tigher on the locking and not have floating locks